network/framework/src/peer_manager/transport.rs

```diff
@@ -257,16 +257,19 @@ where
                 Ok(())
             },
             Err(err) => {
-                warn!(
-                    NetworkSchema::new(&self.network_context)
-                        .remote_peer(&peer_id)
-                        .network_address(&addr),
-                    error = %err,
-                    "{} Outbound connection failed for peer {} at {}: {}",
-                    self.network_context,
-                    peer_id.short_str(),
-                    addr,
-                    err
+                sample!(
```

Comment:
The changes in this file I feel a bit iffy about. These logs are pretty helpful to debug failed network connections. If we absolutely must sample, can we do it every 5 to 10 seconds?